### ISSUE TITLE
gb_mat4_mul_vec4 is broken

### ISSUE THREAD

**Issue by jimon** [gingerBill/gb#11]
Well, technically it's partially broken, but there is a mismatch in row-column major implementations.
To reproduce try this one:

```
gbVec4 translate = gb_vec4(100.0f, 200.0f, 300.0f, 1.0f);
gbVec4 in = gb_vec4(0.0f, 0.0f, 0.0f, 1.0f);
gbVec4 out;
gbMat4 mat;
gb_mat4_translate(&mat, translate.xyz);
gb_mat4_mul_vec4(&out, &mat, in);
printf("%f %f %f %f\n", in.x, in.y, in.z, in.w);
printf("%f %f %f %f\n", out.x, out.y, out.z, out.w);
```

Prints:

```
0.000000 0.000000 0.000000 1.000000
0.000000 0.000000 0.000000 1.000000
```

When it's definitely should print something else in a second line ;)

**Comment by jimon** [comment]
This fixes the issue, but I'm not sure if it's a correct place to fix it? Maybe fix translation matrix instead?

```
void gb_float44_mul_vec4_fixed(gbVec4 *out, float m[4][4], gbVec4 v)
{
    out->x = m[0][0]*v.x + m[1][0]*v.y + m[2][0]*v.z + m[3][0]*v.w;
    out->y = m[0][1]*v.x + m[1][1]*v.y + m[2][1]*v.z + m[3][1]*v.w;
    out->z = m[0][2]*v.x + m[1][2]*v.y + m[2][2]*v.z + m[3][2]*v.w;
    out->w = m[0][3]*v.x + m[1][3]*v.y + m[2][3]*v.z + m[3][3]*v.w;
}
```

**Comment by gingerBill** [comment]
This will be fixed in v0.07.
